src/components/Output/Output.js

```diff
@@ -1,156 +1,130 @@
 import { faPlay, faTerminal } from '@fortawesome/free-solid-svg-icons';
 
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Button } from 'reactstrap';
 import { OUTPUT_ONLY } from '../../constants';
-import OpenPanelButtonContainer from '../common/containers/OpenPanelButtonContainer.js';
-import ViewportAwareButton from '../common/ViewportAwareButton.js';
-import EditorRadio from '../TextEditor/components/EditorRadio.js';
+import OpenPanelButtonContainer from '../common/containers/OpenPanelButtonContainer';
+import ViewportAwareButton from '../common/ViewportAwareButton';
+import EditorRadio from '../TextEditor/components/EditorRadio';
 
 /** --------Props--------
  * None
  */
 
-class Output extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      // used for the refresh button
-      counter: 0,
-      run: 0,
-      showConsole: true,
+const Output = React.memo(
+  ({
+    viewMode,
+    viewOnly,
+    screenHeight,
+    vLanguage,
+    language,
+    code,
+    runResult,
+    updateViewMode,
+    isSmall,
+  }) => {
+    const [counter, setCounter] = useState(0);
+    const [run, setRun] = useState(0);
+    const [showConsole, setShowConsole] = useState(true);
+    const [firstLoad, setFirstLoad] = useState(true);
+
+    useEffect(() => {
+      setFirstLoad(false);
```

Comment:
I don't think the `shouldComponentUpdate` function was refactored correctly. You should pass in a second argument to react memo. Additionally, the useEffect dependency array should correspond to line 43-45 in the old file.